src/org/montclairrobotics/sprocket/drive/Drivetrain.java

```diff
@@ -0,0 +1,37 @@
+package org.montclairrobotics.sprocket.drive;
+
+import org.montclairrobotics.sprocket.geometry.Angle;
+import org.montclairrobotics.sprocket.geometry.Vector;
+import org.montclairrobotics.sprocket.loop.Updatable;
+import org.montclairrobotics.sprocket.pipeline.Pipeline;
+
+public class Drivetrain implements Updatable{
+	private Pipeline<DriveTrainTarget> input;
+	private DriveModule[] modules;
+	
+	public Drivetrain(DriveModule[] modules)
+	{
+		this.input=new Pipeline<DriveTrainTarget>(DriveTrainTarget.ZERO);
+		this.modules=modules;
+	}
+	
+	public void setInput(Pipeline<DriveTrainTarget> input)
+	{
+		this.input=input;
+	}
+
+	@Override
+	public void update() {
+		DriveTrainTarget target=input.get();
+		Vector tgtDirection=target.getDirection();
+		Angle tgtTurn=target.getTurn();
+		for(DriveModule module:modules)
```

Comment:
Again, I feel like all of this sort of stuff should go through a power mapper implementation and shouldn't be hard coded. Not necessarily for the sake of having power mappers be interchangeable, but just for the sake of organisation.

Reply:
Same as above